proof-systems/src/darlin/dlog_acc_marlin/data_structures.rs

```diff
@@ -0,0 +1,333 @@
+use crate::darlin::{MarlinProverKey, MarlinVerifierKey};
+use algebra::{
+    CanonicalDeserialize, CanonicalSerialize, Curve, Read, SerializationError, ToBytes, Write,
+};
+use digest::Digest;
+use marlin::{Proof, IOP};
+use poly_commit::ipa_pc::InnerProductArgPC;
+use poly_commit::{DomainExtendedPolynomialCommitment, PolynomialCommitment};
+
+pub type PC<G, D> = DomainExtendedPolynomialCommitment<G, InnerProductArgPC<G, D>>;
+
+#[derive(Derivative)]
+#[derivative(
+    Clone(bound = ""),
+    Debug(bound = ""),
+    Eq(bound = ""),
+    PartialEq(bound = "")
+)]
+#[derive(CanonicalSerialize, CanonicalDeserialize)]
+pub struct DLogAccMarlinProverKey<G: Curve, D: Digest + 'static>(pub MarlinProverKey<G, PC<G, D>>);
+
+#[derive(Derivative)]
+#[derivative(
+    Clone(bound = ""),
+    Debug(bound = ""),
+    Eq(bound = ""),
+    PartialEq(bound = "")
+)]
+#[derive(CanonicalSerialize, CanonicalDeserialize)]
+pub struct DLogAccMarlinVerifierKey<G: Curve, D: Digest + 'static>(
+    pub MarlinVerifierKey<G, PC<G, D>>,
+);
+
```

Comment:
Do you really need all these functions for DLogAccMarlinProof ? It's just a wrapper for Proof. Can't we re-use some code or saving some functions with the aid of Deref ?